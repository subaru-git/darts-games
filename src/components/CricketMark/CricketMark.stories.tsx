/* eslint-disable import/named */
import React from 'react'
import { ComponentMeta, ComponentStory } from '@storybook/react'
import CricketMark from './CricketMark'

export default {
  title: 'Components/CricketMark',
  component: CricketMark,
} as ComponentMeta<typeof CricketMark>

const Template: ComponentStory<typeof CricketMark> = (args) => <CricketMark {...args} />

export const Default = Template.bind({})
Default.args = {
  count: 3,
}
