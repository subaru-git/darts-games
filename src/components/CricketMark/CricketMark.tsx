import React, { FC } from 'react';
import { useBreakpointValue } from '@chakra-ui/react';
import { css } from '@emotion/react';

type CricketMarkProps = {
  count: number;
};

const CricketMark: FC<CricketMarkProps> = ({ count }) => {
  const isMd = useBreakpointValue({ base: false, md: true });
  return (
    <>
      {count === 3 ? (
        <div css={[threeMarkStyles, isMd ? threeMarkStylesMd : null]} />
      ) : count === 2 ? (
        <div css={[twoMarkStyles, isMd ? twoMarkStylesMd : null]} />
      ) : count === 1 ? (
        <div css={[oneMarkStyles, isMd ? oneMarkStylesMd : null]} />
      ) : null}
    </>
  );
};

const oneMarkStyles = css({
  display: 'inline-block',
  position: 'relative',
  width: '16px',
  height: '16px',
  '&::after': {
    content: '""',
    position: 'absolute',
    top: '50%',
    left: '50%',
    width: '2.5px',
    height: '22px',
    background: 'black',
    borderRadius: '1.25px',
    transform: 'translate(-50%, -50%) rotate(-45deg)',
  },
});
const oneMarkStylesMd = css({
  width: '32px',
  height: '32px',
  '&::after': {
    width: '5px',
    height: '44px',
    borderRadius: '2.5px',
  },
});

const twoMarkStyles = css({
  display: 'inline-block',
  position: 'relative',
  width: '16px',
  height: '16px',
  '&::before, &::after': {
    content: '""',
    position: 'absolute',
    top: '50%',
    left: '50%',
    width: '2.5px',
    height: '22px',
    background: 'black',
    borderRadius: '1.25px',
  },
  '&::before': {
    transform: 'translate(-50%, -50%) rotate(45deg)',
  },
  '&::after': {
    transform: 'translate(-50%, -50%) rotate(-45deg)',
  },
});
const twoMarkStylesMd = css({
  width: '32px',
  height: '32px',
  '&::before, &::after': {
    width: '5px',
    height: '44px',
    borderRadius: '2.5px',
  },
});

const threeMarkStyles = css({
  display: 'inline-block',
  position: 'relative',
  width: '16px',
  height: '16px',
  borderRadius: '100%',
  border: '2.5px solid black',
  '&::before, &::after': {
    content: '""',
    position: 'absolute',
    top: '50%',
    left: '50%',
    width: '2.5px',
    height: '22px',
    background: 'black',
    borderRadius: '1.25px',
  },
  '&::before': {
    transform: 'translate(-50%, -50%) rotate(45deg)',
  },
  '&::after': {
    transform: 'translate(-50%, -50%) rotate(-45deg)',
  },
});
const threeMarkStylesMd = css({
  width: '32px',
  height: '32px',
  border: '5px solid black',
  '&::before, &::after': {
    width: '5px',
    height: '44px',
    borderRadius: '2.5px',
  },
});

export default CricketMark;
