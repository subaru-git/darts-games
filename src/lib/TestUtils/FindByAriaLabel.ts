export const findByAriaLabel = (elements: HTMLElement[], label: string) => {
  return (
    elements.find((button) => button.attributes.getNamedItem('aria-label')?.value === label) ??
    new HTMLElement()
  );
};
