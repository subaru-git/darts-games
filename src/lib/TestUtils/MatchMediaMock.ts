import mediaQuery from 'css-mediaquery'

export const createMatchMedia =
  (width: number) =>
  (query: string): MediaQueryList => ({
    matches: mediaQuery.match(query, { width }),
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })
