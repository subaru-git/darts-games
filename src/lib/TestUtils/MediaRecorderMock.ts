export const MediaRecorderMock = {
  start: jest.fn(),
  ondataavailable: jest.fn(),
  onerror: jest.fn(),
  state: '',
  stop: jest.fn(),
};
