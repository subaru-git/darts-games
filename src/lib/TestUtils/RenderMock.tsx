import { ReactNode } from 'react';
import { ChakraProvider } from '@chakra-ui/react';
import { render as defaultRender } from '@testing-library/react';
import Providers from '@/contexts/Providers';

const render = (children: ReactNode | ReactNode[]) =>
  defaultRender(
    <ChakraProvider>
      <Providers>{children}</Providers>
    </ChakraProvider>,
  );
export { render };
