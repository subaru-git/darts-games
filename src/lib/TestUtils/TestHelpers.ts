import { waitFor as originalWaitFor, waitForOptions } from '@testing-library/react';

const defaultTimeout = process.env.WAIT_FOR_DEFAULT_TIMEOUT ? parseInt(process.env.WAIT_FOR_DEFAULT_TIMEOUT, 10) : 5000;

export async function waitFor(callback: () => void,  options?: waitForOptions): Promise<void> {
  return originalWaitFor(callback, { timeout: defaultTimeout, ...options });
}