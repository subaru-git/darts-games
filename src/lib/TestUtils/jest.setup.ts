import mocksdk from 'firebase-mock';
import failOnConsole from 'jest-fail-on-console';
import { createMatchMedia } from '@/lib/TestUtils/MatchMediaMock';
import { MediaRecorderMock } from '@/lib/TestUtils/MediaRecorderMock';

jest.mock('next/router', () => ({ useRouter: jest.fn().mockReturnValue({ locale: 'en' }) }));
jest.mock('react-camera-pro');
jest.mock('@/firebase/client', () => ({
  analytics: mocksdk.analytics,
  db: mocksdk.firestore,
  storage: mocksdk.storage,
  auth: mocksdk.auth,
  functions: mocksdk.functions,
}));
jest.mock('@firebase/auth');
beforeAll(() => {
  window.matchMedia = createMatchMedia(1100);
  window.MediaRecorder = (jest.fn() as any).mockImplementation(() => MediaRecorderMock);
  global.ResizeObserver = require('resize-observer-polyfill');
});

failOnConsole();
