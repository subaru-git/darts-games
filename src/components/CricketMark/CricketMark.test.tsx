import { ChakraProvider } from '@chakra-ui/react';
import { render } from '@testing-library/react';
import { createMatchMedia } from '../../lib/TestUtils/MatchMediaMock';
import CricketMark from './CricketMark';

test('should rendering', () => {
  window.matchMedia = createMatchMedia(1000);
  const { container } = render(
    <ChakraProvider>
      <CricketMark count={3} />
    </ChakraProvider>,
  );
  expect(container.querySelectorAll('div')).toHaveLength(1);
  expect(container).toMatchSnapshot();
});
