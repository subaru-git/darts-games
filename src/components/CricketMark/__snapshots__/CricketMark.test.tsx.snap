// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendering 1`] = `
<div>
  <div
    class="css-lzrd4v-CricketMark"
  />
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
